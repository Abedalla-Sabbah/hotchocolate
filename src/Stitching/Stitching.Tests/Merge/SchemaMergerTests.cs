using ChilliCream.Testing;
using HotChocolate.Language;
using Xunit;

namespace HotChocolate.Stitching
{
    public class SchemaMergerTests
    {
        [Fact]
        public void MergeSimpleSchemaWithDefaultHandler()
        {
            // arrange
            DocumentNode schema_a =
                Parser.Default.Parse("union Foo = Bar | Baz union A = B | C");
            DocumentNode schema_b =
                Parser.Default.Parse("union Foo = Bar | Baz");

            // act
            DocumentNode schema = SchemaMerger.New()
                .AddSchema("A", schema_a)
                .AddSchema("B", schema_b)
                .Merge();

            // assert
            SchemaSyntaxSerializer.Serialize(schema).Snapshot();
        }

        [Fact]
        public void MergeDemoSchemaWithDefaultHandler()
        {
            // arrange
            DocumentNode schema_a =
                Parser.Default.Parse(
                    FileResource.Open("Contract.graphql"));
            DocumentNode schema_b =
                Parser.Default.Parse(
                    FileResource.Open("Customer.graphql"));

            // act
            DocumentNode schema = SchemaMerger.New()
                .AddSchema("A", schema_a)
                .AddSchema("B", schema_b)
                .Merge();

            // assert
            SchemaSyntaxSerializer.Serialize(schema).Snapshot();
        }
    }
}
