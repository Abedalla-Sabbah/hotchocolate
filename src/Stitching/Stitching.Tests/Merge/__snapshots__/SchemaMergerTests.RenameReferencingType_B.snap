"type B @source(name: \"B\", schema: \"B\") {\n  b1: String\n  b3: String\n}\n\ntype C @source(name: \"C\", schema: \"B\") {\n  c: String\n}\n\ninterface A_C @source(name: \"C\", schema: \"A\") {\n  c: String\n}\n\ntype A @source(name: \"A\", schema: \"A\") {\n  b1: Foo\n}\n\ntype Foo implements A_C @source(name: \"B\", schema: \"A\") {\n  c: String\n}"